pkg/inject/envoy.go

```diff
@@ -249,6 +259,41 @@ func (m *envoyMutator) getCustomEnv(pod *corev1.Pod) (map[string]string, error)
 	return customEnv, nil
 }
 
+type customEnvJsonType map[string]string
```

Comment:
Does this custom type still need to exist, or can `getCustomEnvJson` just deserialize to `map[string]string` directly?

Reply:
The only place I see this type used is in the `getCustomEnvJson` function, which immediately unwraps it to a `map[string]string`.

Generally, you only need to some serialization-time changes between the representative data type and the target data type (for instance, when converting strings to dates). I think that might have been true of the first draft of this where the input data was a list of strings, but now that it's just a map of string->string it's the same as what the target type is. Just passing `map[string]string` to unmarshal should accomplish this fine without the indirection.